src/config/backends/config-fs.ts

```diff
@@ -0,0 +1,231 @@
+import * as _ from 'lodash';
+import { child_process, fs } from 'mz';
+import * as path from 'path';
+
+import {
+	ConfigOptions,
+	DeviceConfigBackend,
+	bootMountPoint,
+	remountAndWriteAtomic,
+} from './backend';
+import * as constants from '../../lib/constants';
+import * as logger from '../../logger';
+import log from '../../lib/supervisor-console';
+
+/**
+ * A backend to handle ConfigFS host configuration
+ *
+ * Supports:
+ * 	- {BALENA|RESIN}_HOST_CONFIGFS_ssdt = value | "value" | "value1","value2"
+ */
+
+type ConfigfsConfig = Dictionary<string[]>;
+
+export class ConfigfsConfigBackend extends DeviceConfigBackend {
+	private readonly SystemAmlFiles = path.join(
+		constants.rootMountPoint,
+		'boot/acpi-tables',
+	);
+	private readonly ConfigFilePath = path.join(bootMountPoint, 'configfs.json'); // use constant for mount path, rename to ssdt.txt
+	private readonly ConfigfsMountPoint = path.join(
+		constants.rootMountPoint,
+		'sys/kernel/config',
+	);
+	private readonly ConfigVarNamePrefix = `${constants.hostConfigVarPrefix}CONFIGFS_`;
+
+	// supported backend for the following device types...
+	public static readonly SupportedDeviceTypes = ['up-board'];
+	private static readonly BootConfigVars = ['ssdt'];
+
+	private stripPrefix(name: string): string {
+		if (!name.startsWith(this.ConfigVarNamePrefix)) {
+			return name;
+		}
+		return name.substr(this.ConfigVarNamePrefix.length);
+	}
+
+	private async listLoadedAcpiTables(): Promise<string[]> {
+		const acpiTablesDir = path.join(this.ConfigfsMountPoint, 'acpi/table');
+		return await fs.readdir(acpiTablesDir);
+	}
+
+	private async loadAML(aml: string): Promise<boolean> {
+		if (!aml) {
+			return false;
+		}
+
+		const amlSrcPath = path.join(this.SystemAmlFiles, `${aml}.aml`);
+		// log to system log if the AML doesn't exist...
+		if (!(await fs.exists(amlSrcPath))) {
+			log.error(`Missing AML for \'${aml}\'. Unable to load.`);
+			if (logger) {
+				logger.logSystemMessage(
+					`Missing AML for \'${aml}\'. Unable to load.`,
+					{ aml, path: amlSrcPath },
+					'Load AML error',
+					false,
+				);
+			}
+			return false;
+		}
+
+		const amlDstPath = path.join(this.ConfigfsMountPoint, 'acpi/table', aml);
+		try {
+			const loadedTables = await this.listLoadedAcpiTables();
+
+			if (loadedTables.indexOf(aml) < 0) {
+				await fs.mkdir(amlDstPath);
+			}
+
+			log.info(`Loading AML ${aml}`);
+			// we use `cat` here as this didn't work when using `cp` and all
+			// examples of this loading mechanism use `cat`.
+			await child_process.exec(
+				`cat ${amlSrcPath} > ${path.join(amlDstPath, 'aml')}`,
+			);
```

Comment:
Any reason not to handle this in nodejs, eg
```ts
fs.createReadStream(amlSrcPath).on('error', ...).pipe(fs.createWriteStream(path.join(amlDstPath, 'aml'))).on('error', ...);
```
it just feel a bit weird to be calling out to another binary to handle this

Reply:
@richbayliss should know the reason exactly, but I remember none of nodes tools seemed to work (streams, async and sync).